UM/FileHandler/WriteFileJob.py

```diff
@@ -19,7 +19,7 @@ class WriteFileJob(Job):
     The writer defines what the result of this job is.
     """
 
-    def __init__(self, writer: Optional[FileWriter], stream: Union[io.BytesIO, io.StringIO], data: Any, mode: int) -> None:
+    def __init__(self, writer: Optional[FileWriter], stream: Union[io.BytesIO, io.StringIO], data: Any, mode: int, writer_args: dict = {}) -> None:
```

Comment:
You shouldn't use a dict like this! It's a bit of a weird thing that python does. So basically what happens here is that it generates a _single_ default dict that is static. So every time the default value is used, it references the same dict. 

Instead, you should do something like:

```suggestion
    def __init__(self, writer: Optional[FileWriter], stream: Union[io.BytesIO, io.StringIO], data: Any, mode: int, writer_args: Optional[Dict] ) -> None:
        if writer_args is None:
            writer_args = {}
```
Also note that i've changed the type of the writer_args to Dict with a capital letter, as that's the one used by typing.

Reply:
> So basically what happens here is that it generates a _single_ default dict that is static. So every time the default value is used, it references the same dict.

Waw, that is something I didn't expect indeed. I'll try to remember.